site/app/models/gradeable/GradeableUtils.php

```diff
@@ -69,20 +73,17 @@ public static function getGradeablesFromCourse(Core $core, Course $course): arra
      * The method loads from the database of all courses and get all gradeables information.
      * Only load once unless the user refreshes the page.
      *
-     * @param Core $core
-     * @param User $user The user to filter gradeables by
      * @return array<string, array<string, Gradeable>|array<string, GradedGradeable>|array<string, Button>>
      * @throws \Exception if a Gradeable failed to load from the database
      */
-    public static function getAllGradeableListFromUserId(Core $core, User $user): array {
+    public static function getAllGradeableListFromUserId(Core $core, User $user, array $courses, array &$calendar_messages): array {
```

Comment:
Is there a reason why you're passing `$calendar_messages` by reference here?  References are rather dangerous to use because it isn't immediately obvious to the user that their array might be getting updated without their knowledge.

Reply:
It does need to be passed reference in order to be changed in "getGradeablesFromCourse". Added a NOTE to clarify.